### PR TITLE
[ome_application_network_proxy] - Enhancement to support check mode,  "proxy_exclusion_list" and password update and idrac_session enhancement (PR-696)

### LINKED ISSUE #644: [FEATURE]: ome_application_network_proxy needs Proxy Exclusion List
**Describe the solution you'd like**
The existing module `ome_application_network_proxy` needs to support managing the Proxy Exclusion List in the Proxy Configuration

**Describe alternatives you've considered**
Manually setting for now

**Additional context**
image

**Community Note**

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) 
  to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, 
  they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### LINKED ISSUE #696: [FEATURE]: Add aliases to the hostname, username and password arguments in the idrac_session module
**Describe the solution you'd like**
Unlike other idrac modules, `idrac_session` module does not support `idrac_ip`, `idrac_user` and `idrac_password` arguments. The request is to add them as aliases to the `hostname`, `username` and `password` arguments in the idrac_session module in order to make it easier and not define multiple variables. I use YAML anchors in my plabook to share values among my tasks, and I don't want to define a new anchor for just idrac_session.

**Describe alternatives you've considered**
Explicitly use the module arguments for idrac_session module.

**Additional context**
Describe any relevant context or add any relevant screenshots about the feature request.

**Community Note**

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) 
  to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, 
  they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by rajshekarp87**
# Description
[ome_application_network_proxy] - Enhancement to support check mode,  "proxy_exclusion_list" and password update, and idrac_session enhancement (PR-696)

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

**Review by Saksham-Nautiyal — APPROVED** [review]

File: plugins/modules/ome_application_network_proxy.py [hunk 1]
```diff
@@ -173,56 +252,102 @@ def validate_check_mode_for_network_proxy(payload_diff, module):
...
+        module (object): The Ansible module object.
+        payload (dict): The payload for the network proxy configuration.
+
+    Returns:
+        dict: The updated payload for the network proxy configuration.
+    """
     current_setting = {}
     if not any(payload):
-        module.fail_json(msg="Unable to configure the proxy because proxy configuration settings are not provided.")
+        module.fail_json(msg="Unable to configure the proxy because proxy configuration settings"
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/ome_application_network_proxy.py]: Move the messages to the global variables

File: plugins/modules/ome_application_network_proxy.py [hunk 2]
```diff
@@ -236,17 +361,24 @@ def main():
             payload = get_payload(module)
             updated_payload = get_updated_payload(rest_obj, module, payload)
             resp = rest_obj.invoke_request("PUT", PROXY_CONFIG, data=updated_payload)
+            response_data = resp.json_data
+            proxy_configuration_details = response_data.copy()
+            if not response_data.get("ProxyExclusionList"):
+                proxy_exclusion_list_list = []
+            else:
+                proxy_exclusion_list_str = response_data.get("ProxyExclusionList")
+                proxy_exclusion_list_list = proxy_exclusion_list_str.strip('[]').strip(']').split(',')
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/ome_application_network_proxy.py]: why are we stripping on '[]' and ']'? splitting on ',' is fine, but we should trim the whitespaces too.

File: plugins/modules/ome_application_network_proxy.py [hunk 3]
```diff
@@ -173,56 +252,102 @@ def validate_check_mode_for_network_proxy(payload_diff, module):
...
         current_setting = resp.json_data
         remove_unwanted_keys(remove_keys, current_setting)
-        diff = any(key in current_setting and val != current_setting[key] for key, val in payload.items())
+        diff = any(key in current_setting and val != current_setting[key] for key, val in
+                   payload.items())
         validate_check_mode_for_network_proxy(diff, module)
         if not diff:
-            module.exit_json(msg="No changes made to proxy configuration as entered values are the same as current "
-                                 "configuration values.")
+            module.exit_json(msg="No changes made to proxy configuration as entered values are "
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/ome_application_network_proxy.py]: Move the error messages to global variables

File: plugins/modules/ome_application_network_proxy.py [hunk 4]
```diff
@@ -173,56 +252,102 @@ def validate_check_mode_for_network_proxy(payload_diff, module):
...
-        "enable_authentication": "EnableAuthentication"
+        "enable_authentication": "EnableAuthentication",
+        "ignore_certificate_validation": "SslCheckDisabled",
+        "proxy_exclusion_list": "ProxyExclusionList"
     }
     backup_params = params.copy()
-    remove_keys = ["hostname", "username", "password", "port", "ca_path", "validate_certs", "timeout"]
+    remove_keys = ["hostname", "username", "password", "port", "ca_path", "validate_certs",
+                   "timeout", "update_password"]
     remove_unwanted_keys(remove_keys, backup_params)
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/ome_application_network_proxy.py:282]: what is the use of it, just want to understand, as we are capturing the output of this functions, if we pass one arguments backup_params, it process that one but not returning any value.

Comment (ABHISHEK-SINHA10) [reply plugins/modules/ome_application_network_proxy.py:282]: okay, understood, here, the argument uses the same memory as the variable. so it's removing the keys.

File: plugins/modules/ome_application_network_proxy.py (hunk 1 above)
Comment (rajshekarp87) [reply plugins/modules/ome_application_network_proxy.py]: Done

File: plugins/modules/ome_application_network_proxy.py (hunk 3 above)
Comment (rajshekarp87) [reply plugins/modules/ome_application_network_proxy.py]: Done

**Review by ABHISHEK-SINHA10 — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Pull request merged**